index.js

```diff
@@ -1,55 +1,47 @@
 const path = require('path');
-const resolve = require('resolve');
-const interpret = require('interpret');
 
-const EXTRE = /^[.]?[^.]+([.].*)$/;
+const extension = require('./lib/extension');
+const normalize = require('./lib/normalize');
+const register = require('./lib/register');
 
-function req (moduleName, cwd) {
-  return require(resolve.sync(moduleName, {basedir: cwd}));
-}
-
-function handleLegacy (moduleName, legacyModuleName, cwd) {
-  try {
-    return req(moduleName, cwd);
-  } catch (err) {
-    try {
-      return req(legacyModuleName, cwd);
-    } catch (__) {
-      // nice error messages
-      err.message = err.message.replace(moduleName, moduleName + '\' or \'' + legacyModuleName);
-      throw err;
-    }
-  }
-}
-
-exports.registerFor = function (filepath, cwd) {
-  var match = EXTRE.exec(path.basename(filepath));
-  if (!match) {
-    return;
-  }
-  var ext = match[1];
+exports.prepare = function (extensions, filepath, cwd) {
+  var option, attempt;
+  var attempts = [];
+  var err;
+  var onlyErrors = false;
+  var ext = extension(filepath);
   if (Object.keys(require.extensions).indexOf(ext) !== -1) {
-    return;
+    return true;
   }
-  var moduleName = interpret.extensions[ext];
-  if (!moduleName) {
-    return;
+  var config = normalize(extensions[ext]);
+  if (!config) {
+    throw new Error('No module loader found for "'+ext+'".');
   }
   if (!cwd) {
     cwd = path.dirname(path.resolve(filepath));
   }
-  var legacyModuleName = interpret.legacy[ext];
-  var config = interpret.configurations[moduleName];
-  var compiler;
-  if (legacyModuleName) {
-    compiler = handleLegacy(moduleName, legacyModuleName, cwd);
-  } else {
-    compiler = req(moduleName, cwd);
+  if (!Array.isArray(config)) {
```

Comment:
That means the branch on line 17 will never be entered because it'll get `if(![undefined])`